banzai_floyds/wavelengths.py

```diff
@@ -281,30 +280,33 @@ def do_stage(self, image):
                 # Estimate 1D distortion with higher order polynomials
                 peaks = identify_peaks(flux_1d, flux_1d_error, self.INITIAL_LINE_WIDTHS[order],
                                        self.MIN_LINE_SEPARATIONS[order])
-                peaks = refine_peak_centers(flux_1d, flux_1d_error, peaks, self.INITIAL_LINE_WIDTHS[order])
-                corresponding_lines = correlate_peaks(peaks, linear_solution, self.INITIAL_LINE_WIDTHS[order],
-                                                      self.MATCH_THRESHOLDS[order])
-                initial_wavelength_solutions.append(estimate_distortion(peaks, corresponding_lines,
-                                                                        image.orders.domains[order], order=4))
+                peaks = np.array(refine_peak_centers(flux_1d, flux_1d_error, peaks, self.INITIAL_LINE_WIDTHS[order]))
+                corresponding_lines = np.array(correlate_peaks(peaks, linear_solution, self.LINES,
+                                                               self.MATCH_THRESHOLDS[order])).astype(float)
+                successful_matches = np.isfinite(corresponding_lines)
+                initial_wavelength_solutions.append(estimate_distortion(peaks[successful_matches],
+                                                                        corresponding_lines[successful_matches],
+                                                                        image.orders.domains[order-1], order=4))
```

Comment:
I'm not sure how we want to handle this `domains[i]` issue. 
I'd like for the order number to directly correspond to a reference in the `domain` rather than rely on a numeric ordering.

Should we redefine `domain` to be a dictionary?